wallet/src/main/java/com/android/identity_credential/wallet/ui/prompt/passphrase/PassphrasePrompt.kt

```diff
@@ -22,23 +68,32 @@ import kotlin.coroutines.resume
  *
  * @param activity the [FragmentActivity] to show the Dialog Fragment via Activity's FragmentManager.
  * @param constraints the constraints for the passphrase.
- * @param checkWeakPassphrase if true, checks and disallows for weak passphrase/PINs and also
- *                            shows a hint if this is the case.
+ * @param title the top-most text of the Prompt.
+ * @param content contains more information about the Prompt.
  */
 suspend fun showPassphrasePrompt(
     activity: FragmentActivity,
     constraints: PassphraseConstraints,
-    checkWeakPassphrase: Boolean,
-): String =
+    title: String,
+    content: String,
+): String? =
     suspendCancellableCoroutine { continuation ->
+
         val passphrasePrompt = PassphrasePrompt(
             constraints = constraints,
-            checkWeakPassphrase = checkWeakPassphrase,
-            onPassphraseEntered = { passphrase ->
+            title = title,
+            content = content,
+            // if dialog is dismissed during anytime other than producing the passphrase
+            onDismissDialog = {
+                // resume with an exception since the dialog isn't successful
+                continuation.resumeWithException(IllegalStateException("Prompt was dismissed"))
```

Comment:
Should probably not throw since we don't want to display "An error occurred" when the user decides to cancel the prompt...

Reply:
Initially I had a sealed class with 3 types, Success, Cancel, and Error that can carry payloads, however during Consent Prompt development you said it was unnecessary and to just throw an exception if user taps on the Cancel button. I followed suit with this prompt.. should I revert both prompts to use a sealed class? or just this prompt? or should I filter through `IllegalStateException` messages that are thrown at the end for keywords such as "User canceled" (taking care to update the message wherever its thrown)? I vote for the sealed class approach to unify a common mechanism for communicating results from each Prompt.

Reply:
But also I think with the current infra in place it would take more effort than its worth to add the sealed class response (vs having the infra to begin with and grow it organically) and the fastest solution for what we have would be to filter through the Exception messages for predefined text "User Canceled" exception messages.